### PR TITLE
Added unit tests for context - Issue 307

### PR DESCRIPTION
**Pull request by Drioastorga**
Added unit test for the context pattern.
Fixed some minor bugs in the context implementation in sequential, native and openmp back-ends.

### CODE REVIEW

**Review by jdgarciauc3m — CHANGES_REQUESTED** [review]
This pull request has conflicts with master.

Please solve them.

File: include/seq/sequential_execution.h [hunk 1]
```diff
@@ -233,6 +233,7 @@ class sequential_execution {
         }
       );
     }
+    std::cout<<"PUSH END"<<std::endl;
```
Comment (jdgarciauc3m) [inline include/seq/sequential_execution.h]: You probably do not want to do this.

**Review by jdgarciauc3m — CHANGES_REQUESTED** [review]
Please, address comments.

**Comment by Drioastorga** [comment]
Fixed.

**Review by jdgarciauc3m — APPROVED** [review]

**Pull request merged**